### PR TITLE
use oc_endpoint_to_string64 to utilize stack

### LINKED ISSUE #529: Optimize memory allocation based on heaptrack results
[heaptrack.txt](https://github.com/iotivity/iotivity-lite/files/12600146/heaptrack.txt)

### PR DESCRIPTION
**Pull request by jkralik**
Fixes #529

### CODE REVIEW

File: include/oc_helpers.h [hunk 1]
```diff
@@ -36,6 +37,13 @@
 extern "C" {
 #endif
 
+typedef struct oc_string64_s
+{
+  struct oc_string64_s *next;
```
Comment (Danielius1922) [inline include/oc_helpers.h:42]: Do you have some future use in mind where `oc_string64_t` is put into an `oc_list_t`? From the uses in this PR we don't need that, so the `next` member is not necessary, in fact we skip the whole type and just use `char[64]`.

File: api/oc_endpoint.c [hunk 2]
```diff
@@ -66,26 +66,50 @@ oc_endpoint_set_di(oc_endpoint_t *endpoint, const oc_uuid_t *di)
   memcpy(endpoint->di.id, di->id, sizeof(di->id));
 }
 
-const char *
-oc_endpoint_flags_to_scheme(unsigned flags)
+bool
+oc_endpoint_flags_to_scheme(unsigned flags, char *buf, size_t *buf_len)
```
Comment (Danielius1922) [inline api/oc_endpoint.c:70]: I'd rather avoid in-out parameters (`buf_len` in this case) if we can help it. I would copy `snprintf` -> take the size of the buffer in arguments, return the number of written bytes or -1 on error.

File: include/oc_helpers.h (hunk 1 above)
Comment (jkralik) [reply include/oc_helpers.h]: removed next + moved to internal header.

File: api/oc_endpoint.c (hunk 2 above)
Comment (jkralik) [reply api/oc_endpoint.c]: fixed

File: include/oc_helpers.h [hunk 3]
```diff
@@ -24,6 +24,7 @@
 #include "oc_export.h"
 #include "util/oc_compiler.h"
 #include "util/oc_list.h"
+#include "util/oc_macros_internal.h"
```
[the remark this answers is not in the document]
Comment (jkralik) [reply include/oc_helpers.h]: removed

**Pull request merged**